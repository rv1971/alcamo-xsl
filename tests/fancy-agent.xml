<?xml version="1.0" encoding="UTF-8"?>

<?xml-stylesheet href="#xslt" type="text/xml"?>

<a:doc
    xmlns:dc="http://purl.org/dc/terms/"
    xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#">
  <xsl:stylesheet
      xmlns="http://www.w3.org/1999/xhtml"
      xmlns:dc="http://purl.org/dc/terms/"
      xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
      xml:id="xslt"
      version="1.0">
    <xsl:import href="../xsl/fancy-html.xsl"/>

    <xsl:template match="/">
      <a:result>
        <xsl:apply-templates select="/a:doc/a:data"/>
     </a:result>
    </xsl:template>

    <xsl:template match="a:item">
      <a:item>
        <xsl:value-of select="@*"/>
        <xsl:value-of select="' '"/>
        <xsl:apply-templates select="@*" mode="a:agent"/>
      </a:item>
    </xsl:template>

    <xsl:template match="dc:*">
      <a:item>
        <xsl:value-of select="."/>
        <xsl:value-of select="' '"/>
        <xsl:apply-templates select="." mode="a:agent"/>
      </a:item>
    </xsl:template>
  </xsl:stylesheet>

  <a:data>
    <dc:creator>no-mail</dc:creator>
    <dc:publisher>alice@example.org</dc:publisher>
    <dc:rightsHolder>mailto:bob@example.org</dc:rightsHolder>
    <a:item dc:creator="Ken &lt;ken@example.org&gt;"/>
    <a:item dc:publisher="http://www.example.com"/>
    <a:item dc:rightsHolder="https://www.example.info/mary"/>
  </a:data>
</a:doc>
