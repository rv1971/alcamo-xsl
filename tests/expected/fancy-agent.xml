<?xml version="1.0"?>
<a:result xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#" xmlns="http://www.w3.org/1999/xhtml" xmlns:dc="http://purl.org/dc/terms/">
  <a:item>no-mail no-mail</a:item>
  <a:item>alice@example.org <a href="mailto:alice@example.org">alice</a></a:item>
  <a:item>mailto:bob@example.org <a href="mailto:bob@example.org">bob</a></a:item>
  <a:item>Ken &lt;ken@example.org&gt; <a href="mailto:ken@example.org">Ken</a></a:item>
  <a:item>http://www.example.com <a href="http://www.example.com">www.example.com</a></a:item>
  <a:item>https://www.example.info/mary <a href="https://www.example.info/mary">www.example.info/mary</a></a:item>
</a:result>
